baybe/surrogates/base.py

```diff
@@ -56,112 +97,250 @@ class Surrogate(ABC, SerialMixin):
-                denotes the "q-batch" shape, and ``d`` is the input dimension. For
-                more details about batch shapes, see: https://botorch.org/docs/batching
+        This method is supposed to be overridden by subclasses to implement their
+        custom parameter scaling logic.
+        """
+        from botorch.models.transforms.input import Normalize
 
-        Returns:
-            The posterior means and posterior covariance matrices of the t-batched
-            candidate points.
+        return Normalize
+
+    @staticmethod
+    def _make_target_scaler_factory() -> type[OutcomeTransform] | None:
+        """Return the scaler factory to be used for target scaling.
+
+        This method is supposed to be overridden by subclasses to implement their
+        custom target scaling logic.
         """
-        import torch
+        from botorch.models.transforms.outcome import Standardize
 
-        # Prepare the input
-        candidates = _prepare_inputs(candidates)
+        return Standardize
 
-        # Evaluate the posterior distribution
-        mean, covar = self._posterior(candidates)
+    def _make_input_scaler(self, searchspace: SearchSpace) -> ColumnTransformer:
+        """Make and fit the input scaler for transforming computational dataframes."""
+        # Create a composite scaler from parameter-wise scaler objects
+        mapping: dict[tuple[int, ...], InputTransform] = {}
+        for p in searchspace.parameters:
+            if (factory := self._make_parameter_scaler_factory(p)) is None:
+                continue
+            idxs = searchspace.get_comp_rep_parameter_indices(p.name)
+            transformer = factory(len(idxs))
+            mapping[idxs] = transformer
+        scaler = ColumnTransformer(mapping)
 
-        # Apply covariance transformation for marginal posterior models
-        if not self.joint_posterior:
-            # Convert to tensor containing covariance matrices
-            covar = torch.diag_embed(covar)
+        # Fit the scaler to the parameter bounds
+        scaler.fit(to_tensor(searchspace.comp_rep_bounds))
 
-        # Add small diagonal variances for numerical stability
-        covar.add_(torch.eye(covar.shape[-1]) * _MIN_VARIANCE)
+        return scaler
 
-        return mean, covar
+    def _make_output_scaler(
+        self, objective: Objective, measurements: pd.DataFrame
+    ) -> OutcomeTransform | _NoTransform:
+        """Make and fit the output scaler for transforming computational dataframes."""
+        if (factory := self._make_target_scaler_factory()) is None:
+            return _IDENTITY_TRANSFORM
 
-    @abstractmethod
-    def _posterior(self, candidates: Tensor) -> tuple[Tensor, Tensor]:
-        """Perform the actual posterior evaluation logic.
+        # TODO: Multi-target extension
+        scaler = factory(1)
+
+        # TODO: Consider taking into account target boundaries when available
```

Comment:
That comment was basically copied from the GP class where it existed basically since Day 1, even before we had other surrogates. It reminds us that we have never considered using the target bounds, which are available in several cases.

So in the current state, the comment is:
* in the GP class, because it currently has it's own scaling
* in the surrogate base class, because the same question applied to all surrogates.

So bottom line: I'm not exactly sure what you are asking here 😄 Do you want me to move the comment? Or implement it (<-- definitely out of PR scope)? ...

Reply:
im certainly not asking for implementation

if bound scaling is part of targets, surely it would be encompassed in their transform method

hence, why is this comment here

Reply:
These are two different concerns, actually. Right now, our only approach to target scaling is in the context of the desirability objective. However, using target bounds for surrogate normalization is a different thing, and we should clean this up at some point. Right now, you could have a single target with defined bounds, but unless you also manually add a transformation, no scaling will be applied. Still, the bound information can be really useful for the surrogate, exactly in the same way as we use the bounds of the parameters. Ultimately, this is still all the same old topic as the one addressed in #114. Now that we have a clean approach to scaling, we can finally fix this part as well.

Since I don't think we'll get to a conclusion here, can we just keep the comments and address all of it in October when I'm back? Even if this means to just ignore it, remove the comments and close the PR. But I suspect that the outcome of the discussion will be different.

Reply:
not really, the targets also have a `.transform`, naively I'd say something with their bounds happens here too, but anyway seems the issue needs to be thought through a bit more so leave the comment